dashboard/app/models/pd/workshop_daily_survey.rb

```diff
@@ -62,26 +64,45 @@ def set_day_from_form_id
       :day
     )
 
-    VALID_DAYS = (0..5).freeze
+    # Different categories have different valid days
+    VALID_DAYS = {
+      LOCAL_CATEGORY => (0..5).to_a.freeze,
+      ACADEMIC_YEAR_1_CATEGORY => [1].freeze,
+      ACADEMIC_YEAR_2_CATEGORY => [1].freeze,
+      ACADEMIC_YEAR_3_CATEGORY => [1].freeze,
+      ACADEMIC_YEAR_4_CATEGORY => [1].freeze,
+      ACADEMIC_YEAR_1_2_CATEGORY => [1, 2].freeze,
+      ACADEMIC_YEAR_3_4_CATEGORY => [1, 2].freeze,
+    }
 
-    validates_inclusion_of :day, in: VALID_DAYS
+    validate :day_for_subject
 
-    def self.get_form_id_for_day(day)
-      get_form_id 'local', "day_#{day}"
+    def self.get_form_id_for_day_and_subject(subject, day)
```

Comment:
I would prefer the order of params and the name of the method to match - so either flipping the params for changing the method to `get_form_id_for_subject_and_day`